### DISCUSSION TITLE
Using an e-mail validation API

### DISCUSSION THREAD

**Discussion by tillathenun** [mplodowski/formbuilder-plugin-public#31] [General]
I'm just looking for some advice if possible, please.

My client is asking that I use an e-mail validation API like [Zerobounce](https://www.zerobounce.net/email-validation-api.html).

I can do the validation inside the `formBuilder.beforeSendMessage` event of course, but then it'd fail silently and I'd like if possible to give the user a validation message while they're entering their e-mail address.

What's the best way of achieving what I want?

**Comment by mplodowski** [comment]
I would create custom plugin. Then I would create custom validation rule. Remember to cache the API result for validated email, so you don't fire it on every form validation request.

https://docs.octobercms.com/3.x/extend/services/validation.html#custom-validation-rules

With this approach you should be able to use your validation like you normally validate fields with formbuilder plugin.

**Reply by tillathenun** [reply]
Thanks very much. I had no idea custom validation rules were a thing! Great tip with the cache, too.

Much appreciated!
